### DISCUSSION TITLE
Is Ory Kratos support for hasher $argon2i instead using $argon2id

### DISCUSSION THREAD

**Discussion by FerdinaKusumah** [ory/kratos#2373] [Q&A] [answered]
Hello,

I want to migrate all our user data to ory kratos, our current password is hashed in `$argon2i` not `$argon2id` when read from documentation ory kratos is supported [this](https://www.ory.sh/docs/guides/manage-identities/import-identity) type of algoritm.

payload will be like this
```
{
    "schema_id": "...",
    "traits": {...},
    "verifiable_addresses": [...],
    "credentials": {
        "password": {
            "config": {
                "hashed_password": "$argon2i$v=19$m=512,t=2,p=2$WURJbGFLVDRxamFk$LI8ZgzyyjpP017eNtFYr8A"
            }
        }
    }
}
```

But when i'm trying to insert this hash password system say
```
{
    "error": {
        "code": 400,
        "status": "Bad Request",
        "request": "bf9ae354-8d5f-94e2-ae7a-f8a5ed7b7884",
        "reason": "The imported password does not match any known hash format. For more information see https://www.ory.sh/dr/2",
        "message": "The request was malformed or contained invalid parameters"
    }
}
```
Is there any config that i need to setup for setup this algorithm ?

Please kindly help me how to support this particular algorithm to insert into ory kratos.
Thank you very much

**Comment by aeneasr** [comment, marked as answer]
It’s possible that we don’t support Argon2i yet, but it would be awesome if we did! If you’re down for a PR please go ahead :)
